.github/workflows/windows-alt.yml

```diff
@@ -63,13 +63,56 @@ jobs:
           options: |
             CMAKE_SYSTEM_NAME=Windows \
             CMAKE_SYSTEM_PROCESSOR=x86_64 \
-            CMAKE_MAKE_PROGRAM=ninja.exe \
             CMAKE_BUILD_TYPE=Release \
       - name: Build Project
         run: cmake --build ./build --target all
       - name: Run tests
         run: cmake --build ./build --target run_tests
-  clang-cl:
+  clang-cl-msbuild:
+    if: github.repository_owner == 'aws'
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - x64
+          - x64_arm64
+    runs-on: windows-latest
+    env:
+      CMAKE_GENERATOR: "Visual Studio 17 2022"
+      CMAKE_GENERATOR_TOOLSET: "ClangCL,host=x64"
+    steps:
+      - if: ${{ matrix.target  == 'x64' }}
+        name: Install NASM
+        uses: ilammy/setup-nasm@v1.5.1
+      - name: Checkout
+        uses: actions/checkout@v4
+      - uses: TheMrMilchmann/setup-msvc-dev@v3
+        with:
+          arch: ${{ matrix.target }}
+      - if: ${{ matrix.target  == 'x64' }}
+        name: Setup CMake
+        uses: threeal/cmake-action@v1.3.0
+        with:
+          options: |
+            CMAKE_BUILD_TYPE=Release \
+      - if: ${{ matrix.target  == 'x64_arm64' }}
+        name: Setup CMake
+        uses: threeal/cmake-action@v1.3.0
+        with:
+          options: |
+            CMAKE_GENERATOR_PLATFORM=ARM64 \
+            CMAKE_SYSTEM_NAME=Windows \
+            CMAKE_SYSTEM_PROCESSOR=ARM64 \
+            CMAKE_BUILD_TYPE=Release \
+      - name: Build Project
+        run: cmake --build ./build --target all_tests
+      - if: ${{ matrix.target  == 'x64' }}
+        name: Run crypto_test
+        # MSVC places the tests executables in a different location.
+        # With MSVC, the "run_tests" target cannot locate the tests.
+        # We run "crypto_test" as a sanity check.
```

Comment:
MSVC inserts the build type (e.g., "Debug" or "Release") into the path right above the binary.  So instead of [`"crypto/crypto_test"`](https://github.com/aws/aws-lc/blob/main/util/all_tests.json#L17), the path would need to be `"crypto/Debug/crypto_test"` or `"crypto/Release/crypto_test"`.

There might be a way to alter the artifact path whenever we see that the Generator matches "Visual Studio", but it's not been an issue prior to now. So I don't think it's worth the effort to pursue it.